libs/block-editor/src/lib/extensions/dragHandler.extention.ts

```diff
@@ -0,0 +1,156 @@
+import { Extension } from '@tiptap/core';
+import { NodeSelection, Plugin, PluginKey } from 'prosemirror-state';
+import { EditorView } from 'prosemirror-view';
+
+export const DragHandler = Extension.create({
+    name: 'dragHandler',
+
+    addProseMirrorPlugins() {
+        let nodeToBeDragged = null;
+        const WIDTH = 24;
+        let dragHandler;
+
+        function createRect(rect) {
+            if (rect == null) {
+                return null;
+            }
+            let newRect = {
+                left: rect.left + document.body.scrollLeft,
+                top: rect.top + document.body.scrollTop,
+                width: rect.width,
+                height: rect.height,
+                bottom: 0,
+                right: 0
+            };
+            newRect.bottom = newRect.top + newRect.height;
+            newRect.right = newRect.left + newRect.width;
+            return newRect;
+        }
+
+        function absoluteRect(element: HTMLElement): ClientRect {
+            return createRect(element.getBoundingClientRect());
+        }
+
+        function removeNode(node) {
+            if (node && node.parentNode) {
+                node.parentNode.removeChild(node);
+            }
+        }
+
+        function blockPosAtCoords(coords, view) {
+            let pos = view.posAtCoords(coords);
+            if (pos) {
+                let node = getDirectChild(view.nodeDOM(pos.inside));
+                if (node && node.nodeType === 1) {
+                    let desc = view.docView.nearestDesc(node, true);
+                    if (!(!desc || desc === view.docView)) {
+                        return desc.posBefore;
+                    }
+                }
+            }
+            return null;
+        }
+
+        function dragStart(e, view: EditorView) {
+            if (!e.dataTransfer) return;
+            let coords = { left: e.clientX + 50, top: e.clientY };
+            let pos = blockPosAtCoords(coords, view);
+            if (pos != null) {
+                view.dispatch(
+                    view.state.tr.setSelection(NodeSelection.create(view.state.doc, pos))
+                );
+                let slice = view.state.selection.content();
+                e.dataTransfer.clearData();
+                debugger;
+                e.dataTransfer.setDragImage(nodeToBeDragged, 10, 10);
+                view.dragging = { slice, move: true };
+            }
+        }
+
+        // Get the direct child of the Editor.
+        function getDirectChild(node) {
+            while (node && node.parentNode) {
+                if (
+                    node.classList?.contains('ProseMirror') ||
+                    node.parentNode.classList?.contains('ProseMirror')
+                ) {
+                    break;
+                }
+                node = node.parentNode;
+            }
+            return node;
+        }
+
+        // Check if not has content.
```

Comment:
If the source or the target doesn't have content?

Reply:
then we do not display the handler because there is nothing to drag